commit 7ef20f058b98950c595b92e767efd3ce630940e0
Author: Chris Burroughs <chris.burroughs@joyent.com>
Date:   2019-03-05T18:42:30+00:00 (7 months ago)
    
    TOOLS-2192 buildimage target fails with non-default BUILDIMAGE_VERSION value
    Reviewed by: Tim Foster <tim.foster@joyent.com>
    Approved by: Tim Foster <tim.foster@joyent.com>

diff --git a/tools/mk/Makefile.defs b/tools/mk/Makefile.defs
index bb6370a..fa3569e 100644
--- a/tools/mk/Makefile.defs
+++ b/tools/mk/Makefile.defs
@@ -192,9 +192,9 @@ STAMP_BUILDIMAGE_PREP := $(MAKE_STAMPS_DIR)/buildimage-prep
 # Metadata needed by buildimage to construct the image.
 # This gets used when generating the image's manifest file.
 #
-BUILDIMAGE_VERSION	= $(STAMP)
+BUILDIMAGE_VERSION	?= $(STAMP)
 BUILDIMAGE_STAGEDIR = /tmp/buildimage-$(NAME)-$(STAMP)
-BUILDIMAGE_MF           = \
+BUILDIMAGE_MF           ?= \
     {"name": "$(BUILDIMAGE_NAME)",\
     "description": "$(BUILDIMAGE_DESC)",\
     "version": "$(BUILDIMAGE_VERSION)",\
diff --git a/tools/mk/Makefile.targ b/tools/mk/Makefile.targ
index 3c9ac4c..6744fe8 100644
--- a/tools/mk/Makefile.targ
+++ b/tools/mk/Makefile.targ
@@ -439,10 +439,10 @@ buildimage: stamp-buildimage-prep release $(AGENTS:%=%-prebuilt)
 		-m '$(BUILDIMAGE_MF)' \
 		$(BUILDIMAGE_PKGSRC_ARGS) \
 		-P $(NAME)-zfs
-	cp /tmp/$(NAME)-zfs-$(STAMP).zfs.gz $(ENGBLD_BITS_DIR)/$(NAME)
-	cp /tmp/$(NAME)-zfs-$(STAMP).imgmanifest $(ENGBLD_BITS_DIR)/$(NAME)
-	pfexec rm /tmp/$(NAME)-zfs-$(STAMP).zfs.gz
-	pfexec rm /tmp/$(NAME)-zfs-$(STAMP).imgmanifest
+	cp /tmp/$(NAME)-zfs-$(BUILDIMAGE_VERSION).zfs.gz $(ENGBLD_BITS_DIR)/$(NAME)
+	cp /tmp/$(NAME)-zfs-$(BUILDIMAGE_VERSION).imgmanifest $(ENGBLD_BITS_DIR)/$(NAME)
+	pfexec rm /tmp/$(NAME)-zfs-$(BUILDIMAGE_VERSION).zfs.gz
+	pfexec rm /tmp/$(NAME)-zfs-$(BUILDIMAGE_VERSION).imgmanifest
 	pfexec rm -rf $(BUILDIMAGE_STAGEDIR)
 
 #
